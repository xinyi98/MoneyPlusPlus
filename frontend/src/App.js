import { Switch, Route, Redirect } from 'react-router-dom';
import { useState } from 'react';
import TransactionPage from './Pages/TransactionPage';
import SummaryPage from './Pages/SummaryPage';
import CurrencyPage from './Pages/CurrencyPage';
import Header from './Components/Header';
import TipsPage from './Pages/TipsPage';
import LoginPage from './Pages/LoginPage';
import './App.css';
import SignupPage from './Pages/SignupPage';

function App() {
  const [login, setLogin] = useState(!!window.sessionStorage.getItem('MoneyPlusPlusToken'));
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  return (
    <div className="App">
      {login ? (
        <div>
          <Header setLogin={setLogin} username={username} />
          <Switch>
            <Route exact path="/transaction">
              <TransactionPage />
            </Route>
            <Route exact path="/summary">
              <SummaryPage />
            </Route>
            <Route exact path="/currency">
              <CurrencyPage />
            </Route>
            <Route exact path="/tips">
              <TipsPage />
            </Route>
            <Route path="/*">
              <Redirect to="/transaction" />
            </Route>
          </Switch>
        </div>
      ) : (
        <Switch>
          <Route exact path="/signup">
            <SignupPage />
          </Route>
          <Route exact path="/">
            <LoginPage
              setLogin={setLogin}
              username={username}
              password={password}
              setUsername={setUsername}
              setPassword={setPassword}
            />
          </Route>
          <Route path="/*">
            <Redirect to="/" />
          </Route>
        </Switch>
      )}
    </div>
  );
}

export default App;
