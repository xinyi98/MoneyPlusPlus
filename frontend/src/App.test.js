import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('snapshot with no content matches', () => {
  const tree = renderer
    .create(
      <Router>
        <App />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
