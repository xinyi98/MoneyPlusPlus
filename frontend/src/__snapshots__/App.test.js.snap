// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with no content matches 1`] = `
<div
  className="App"
>
  <div
    className="login"
  >
    <div
      className="backgroundReactangleContainer"
    >
      <img
        alt="BackgroundRectangle"
        className="backgroundRecatngle"
        src="loginRectangle.svg"
      />
    </div>
    <div
      className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <form
        className="makeStyles-form-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          onInput={[Function]}
          style={
            Object {
              "margin": "15px 0px",
              "width": "100%",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="username-textfield"
            id="username-textfield-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="username-textfield"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-6"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          onInput={[Function]}
          style={
            Object {
              "margin": "15px 0px",
              "width": "100%",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="password-textfield"
            id="password-textfield-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="password-textfield"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-6"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-2 MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
        <div
          className="signUpText"
          style={
            Object {
              "padding": "5px",
              "textAlign": "center",
            }
          }
        >
          <span>
             Don't have an account ? 
          </span>
          <a
            href="/signup"
            onClick={[Function]}
          >
             Sign Up Here 
          </a>
        </div>
        <h1
          className="loginTitle"
        >
          Hello, 
          <br />
           Welcome back
        </h1>
      </form>
    </div>
    <div
      className="backgroundImageContainer"
    >
      <img
        alt="Background"
        className="backgroundImage"
        src="loginImage.svg"
      />
    </div>
  </div>
</div>
`;
